### PR TITLE
Revamp badge

### PR DESCRIPTION
**Pull request by joneff**

### CODE REVIEW

File: packages/default/scss/utils/_position.scss [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            left: auto;
+            right: 0;
+        }
+    }
+    .k-bottom-end,
+    .k-pos-bottom-end {
+        bottom: 0;
+        right: 0;
+
+        .k-rtl &,
```
Comment (elena-gancheva) [inline packages/default/scss/utils/_position.scss:51]: `&[dir="rtl"],` should be added here to support angular rtl mode

File: packages/default/scss/utils/_position.scss [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            right: 0;
+        }
+    }
+    .k-top-end,
+    .k-pos-top-end {
+        top: 0;
+        right: 0;
+
+        .k-rtl &,
+        [dir="rtl"] & {
```
Comment (elena-gancheva) [inline packages/default/scss/utils/_position.scss:30]: same here

File: packages/default/scss/utils/_position.scss [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            left: 0;
+        }
+    }
+    .k-bottom-start,
+    .k-pos-bottom-start {
+        bottom: 0;
+        left: 0;
+
+        .k-rtl &,
+        [dir="rtl"] & {
```
Comment (elena-gancheva) [inline packages/default/scss/utils/_position.scss:41]: same here

File: packages/default/scss/badge/_layout.scss [hunk 4]
```diff
@@ -123,73 +127,51 @@
...
             }
         }
 
-        .k-rtl .k-badge#{$infix},
-        [dir="rtl"] .k-badge#{$infix} {
-            &-top-start {
-                left: auto;
-                right: 0;
+        .k-rtl .k-badge-#{$placement},
+        [dir="rtl"] .k-badge-#{$placement} {
```
Comment (elena-gancheva) [inline packages/default/scss/badge/_layout.scss:158]: `&[dir="rtl"],`

File: packages/default/scss/utils/_position.scss [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        .k-pos-#{$side} { #{$side}: 0; } // sass-lint:disable-line brace-style
+    }
+
+    .k-top-start,
+    .k-pos-top-start {
+        top: 0;
+        left: 0;
+
+        .k-rtl &,
+        [dir="rtl"] & {
```
Comment (elena-gancheva) [inline packages/default/scss/utils/_position.scss:19]: `&[dir="rtl"],`

File: packages/default/scss/badge/_theme.scss [hunk 6]
```diff
@@ -2,7 +2,7 @@
 
     // Solid badges
     @each $name, $color in $theme-colors {
```
Comment (elena-gancheva) [inline packages/default/scss/badge/_theme.scss:4]: [question] are we going to add `inherit` as a value for the themeColor prop?

Comment (joneff) [reply packages/default/scss/badge/_theme.scss:4]: the badge is inherit by deault

**Pull request merged**